app/src/main/java/fi/sabriina/urbanhuikka/viewmodel/GameStateViewModel.kt

```diff
@@ -228,6 +229,17 @@ class GameStateViewModel (private val repository: GameStateRepositoryInterface):
         repository.setPointsToWin(points)
     }
 
+    suspend fun drawAssistingPlayer() : Player {
+        var index =  Random.nextInt(playerList.size)
+
+        while (index == currentPlayerIndex) {
+            index =  Random.nextInt(playerList.size)
+        }
+
+        repository.updateAssistingPlayerIndex(index)
+        return playerList[index]
```

Comment:
Random draw could also be accomplished with `playerList.random()`, which could look simpler here.